homeassistant/components/github/__init__.py

```diff
@@ -64,12 +65,39 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
         hass.data[DOMAIN][repository] = coordinators
 
+    await async_cleanup_device_registry(hass=hass, entry=entry)
+
     hass.config_entries.async_setup_platforms(entry, PLATFORMS)
     entry.async_on_unload(entry.add_update_listener(async_reload_entry))
 
     return True
 
 
+async def async_cleanup_device_registry(
```

Comment:
This can be a `@callback` since we don't await inside.

Reply:
https://github.com/home-assistant/core/pull/64481